snowpack/src/build/build-pipeline.ts

```diff
@@ -173,26 +174,34 @@ async function runPipelineTransformStep(
         logger.debug(`✔ transform() success [${debugPath}]`, {name: step.name});
         if (typeof result === 'string' || Buffer.isBuffer(result)) {
           // V2 API, simple string variant
-          output[destExt].code = result;
-          output[destExt].map = undefined;
+          if (destBuildFile.code !== result) {
+            destBuildFile.map = undefined;
+            if (config.buildOptions.sourceMaps && (destExt === '.js' || destExt === '.css')) {
+              logger.warn(`source-map file not found in [${debugPath}]`, {name: step.name});
```

Comment:
Technically you could still have inline sourcemaps which wouldn't have an external `map`